features/storage/filesystem/FileSystem.h

```diff
@@ -71,18 +71,44 @@ class FileSystem : public FileSystemLike {
      */
     static FileSystem *get_default_instance();
 
+    /** Initializes the file system
+     *
+     *  The init function may be called multiple times recursively as long
+     *  as each init has a matching deinit. If not called, the file system
+     *  will be mounted on the first file system operation.
+     *
+     *  @return         0 on success, negative error code on failure
+     */
+    virtual int init();
+
+    /** Deinitializes the file system
+     *
+     *  @return         0 on success, negative error code on failure
+     */
+    virtual int deinit();
+
+    /** Reformats the underlying filesystem
```

Comment:
Should we specify explicitly that file system will be initialised and mounted as part of reset call?